src/user/user.service.ts

```diff
@@ -79,29 +80,34 @@ export class UserService {
   //***************Find All USERS**********
 
   findAll() {
-    return this.prisma.user.findMany();
+    const where = { deletedAt: null };
+    return this.prisma.user.findMany({
+      where: where,
+    });
   }
 
   // **********Find A USER ***********
   findOne(id: string) {
-    return this.prisma.user.findUnique({
-      where: { id: Number(id) },
+    const where = { id: Number(id), deletedAt: null };
+    return this.prisma.user.findFirst({
+      where: where,
     });
   }
 
-  // ***********Update user************(not done)
-  updateUser(id: string, user: UserDto) {
+  // ***********Update user************
+  updateUser(id: string, user: UserUpdateDto) {
     return this.prisma.user.update({
-      
       where: { id: Number(id) },
       data: user,
     });
   }
 
   // ***********Delete user **********
   deleteUser(id: string) {
-    return this.prisma.user.delete({
+    const user = { deletedAt: new Date() };
+    return this.prisma.user.update({
       where: { id: Number(id) },
```

Comment:
```suggestion
      where: { id: id },
```
if you uuid so it will return string of id